presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java

```diff
@@ -508,33 +508,46 @@ public Signature resolveFunction(QualifiedName name, List<TypeSignature> paramet
         }
 
         // search for coerced matches
-        List<Signature> coercedMatches = new ArrayList<>();
+        List<SqlFunction> coercedCandidates = new ArrayList<>();
+        Signature firstCoercedMatch = null;
```

Comment:
Isn't this the first element of the `coercedCandidates` list? Why do we need a separate variable?